src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -302,16 +304,27 @@ function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourceLoade
 
     // Wire-up callbacks for satellite assemblies. Blazor will call these as part of the application
     // startup sequence to load satellite assemblies for the application's culture.
-    window['Blazor']._internal.getSatelliteAssemblies = (culturesToLoadDotNetArray: System_Array<System_String>) : System_Object =>  {
+    window['Blazor']._internal.getSatelliteAssemblies = (culturesToLoadDotNetArray: System_Array<System_String>): System_Object => {
       const culturesToLoad = BINDING.mono_array_to_js_array<System_String, string>(culturesToLoadDotNetArray);
       const satelliteResources = resourceLoader.bootConfig.resources.satelliteResources;
+      const applicationCulture = resourceLoader.startOptions.applicationCulture || (navigator.languages && navigator.languages[0]);
+
+      if (resourceLoader.bootConfig.icuDataMode == ICUDataMode.Default && culturesToLoad && culturesToLoad[0] !== applicationCulture) {
+        // We load an initial icu file based on the browser's locale. However if the application's culture requires a different set, flag this as an error.
+        if (resourceLoader.bootConfig.debugBuild) {
+          throw new Error('This application\'s globalization settings requires using the combined globalization data file. ' +
+          'To configure your application to use this data file, set <BlazorWebAssemblyLoadAllGlobalizationData>true</BlazorWebAssemblyLoadAllGlobalizationData> in the application\'s project file. ');
+        } else {
+          throw new Error('Globalization data files required to correctly execute this application are missing.');
+        }
```

Comment:
Not sure it's useful to distinguish debug/prod cases here, since in both cases only the application developer will see this or do anything about it. Also it would be good to clarify *why* the situation has occurred. Suggestion:

```suggestion
        throw new Error('To change culture dynamically during startup, set <BlazorWebAssemblyLoadAllGlobalizationData>true</BlazorWebAssemblyLoadAllGlobalizationData> in the application\'s project file.');
```

Is it true that this only happens due to dynamic change of thread culture? Could it ever happen because, for example, the browser says it's in a culture that .NET doesn't even know about the existence of? If the latter could ever happen that's a bit of a bearpit - hope it's not an error by default!

Reply:
> Could it ever happen because, for example, the browser says it's in a culture that .NET doesn't even know about the existence of?

From what I have tried, this is not the case. .NET creates a culture using the value specified by browser \ emscripten regardless of it's known. You get invariant assets, but do not see this error unless the culture is tweaked in .NET code.